crates/bevy_math/src/curve/sample_curves.rs

```diff
@@ -0,0 +1,367 @@
+//! Sample-interpolated curves constructed using the [`Curve`] API.
+
+use super::cores::{EvenCore, EvenCoreError, UnevenCore, UnevenCoreError};
+use super::{Curve, Interval};
+
+use crate::StableInterpolate;
+use core::any::type_name;
+use core::fmt::{self, Debug};
+
+#[cfg(feature = "bevy_reflect")]
+use bevy_reflect::{utility::GenericTypePathCell, Reflect, TypePath};
+
+const THIS_MODULE: &str = "bevy_math::curve::sample_curves";
+const THIS_CRATE: &str = "bevy_math";
+
+/// A curve that is defined by explicit neighbor interpolation over a set of evenly-spaced samples.
+#[derive(Clone)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Should we `#[reflect(Serialize, Deserialize)]`? Same goes for the other types.

Reply:
Oh right, I always forget how much generic types mess up the type data registrations haha

Reply:
Yeah; on the other hand, they have a tendency to make them less useful as well, so 🤷